### PR TITLE
getDigParams: Fix selecting the lowest digging time

### PR DESCRIPTION
**Pull request by HybridDog**
* Tidy up code
* Fix selecting the best fitting time: time was compared with result_time before dividing it by the level difference
* Remove dead code (time from last punch)

### CODE REVIEW

File: src/tool.cpp [hunk 1]
```diff
@@ -110,40 +107,42 @@ DigParams getDigParams(const ItemGroupList &groups,
...
-		result_time /= f;
-		result_wear /= f;
+		if (f > 1.0) {
+			result_time /= f;
+			result_wear /= f;
+		}
 	}
 
-	u16 wear_i = 65535.*result_wear;
+	u16 wear_i = 0xffff * result_wear;
```
Comment (SmallJoker) [inline src/tool.cpp]: `U16_MAX`

Comment (HybridDog) [inline src/tool.cpp]: fixed

File: src/tool.cpp [hunk 2]
```diff
@@ -110,40 +107,42 @@ DigParams getDigParams(const ItemGroupList &groups,
...
-				result_diggable = true;
-				int leveldiff = cap.maxlevel - level;
-				result_time = time / MYMAX(1, leveldiff);
-				if(cap.uses != 0)
-					result_wear = 1.0 / cap.uses / pow(3.0, (double)leveldiff);
-				else
-					result_wear = 0;
-				result_main_group = name;
-			}
+		bool time_exists = cap.getTime(itemgroup_get(groups, groupname), &time);
```
Comment (nerzhul) [inline src/tool.cpp]: no need to create variable, just call this into condition

Comment (SmallJoker) [inline src/tool.cpp]: It's not entirely clear what `getTime` returns, thus it's better to keep it this way. Makes it easier to understand and the compiler would optimize this anyway.

Comment (HybridDog) [inline src/tool.cpp]: ok

File: src/tool.cpp [hunk 3]
```diff
@@ -110,40 +107,43 @@ DigParams getDigParams(const ItemGroupList &groups,
...
-	//infostream<<"result_time="<<result_time<<std::endl;
-	//infostream<<"result_wear="<<result_wear<<std::endl;
 
-	if(time_from_last_punch < tp->full_punch_interval){
+	if (time_from_last_punch < tp->full_punch_interval) {
 		float f = time_from_last_punch / tp->full_punch_interval;
-		//infostream<<"f="<<f<<std::endl;
-		result_time /= f;
-		result_wear /= f;
+		if (f > 1.0) {
```
Comment (SmallJoker) [inline src/tool.cpp]: This is wrong. It can't be `> 1` since `last_punch < punch_interval`

Comment (HybridDog) [inline src/tool.cpp]: You're right. Anyway, l just noticed (```$ ag -r -A2 getDigParams```) that it's unused code.
l'll remove it in a separate commit.

**Comment by HybridDog** [comment]
When can this be merged?

File: src/script/lua_api/l_util.cpp [hunk 4]
```diff
@@ -219,18 +219,14 @@ int ModApiUtil::l_write_json(lua_State *L)
 	return 1;
 }
 
-// get_dig_params(groups, tool_capabilities[, time_from_last_punch])
+// get_dig_params(groups, tool_capabilities)
 int ModApiUtil::l_get_dig_params(lua_State *L)
```
Comment (SmallJoker) [inline src/script/lua_api/l_util.cpp]: Mind documenting `get_hit_params` and `get_dig_params` in the Lua API or does that rather belong into a new pull/issue?

Comment (HybridDog) [inline src/script/lua_api/l_util.cpp]: lt belongs to another PR.

**Comment by paramat** [comment]
> l put the first two in one commit because the code is changed in the same place.

Code cleanup shold still be a separate commit even if it is acting on the same code, mixiing cleanups with other changes makes review harder and creates a more confusing history.

**Comment by paramat** [comment]
Thanks, perhaps the last 2 can be squashed? (this could be done on merge).

**Comment by HybridDog** [comment]
> this could be done on merge

ok

**Comment by HybridDog** [comment]
All checks have failed

**Comment by sfan5** [comment]
Only thing that failed is an unimportant unittest on OS X

**Comment by HybridDog** [comment]
>this could be done on merge

ok

**Comment by HybridDog** [comment]
please feedback

**Comment by HybridDog** [comment]
rebased

**Comment by paramat** [comment]
@SmallJoker still +1? If so this can be merged.

**Comment by SmallJoker** [comment]
@paramat Yes.

**Comment by paramat** [comment]
d7c1f6c92ec612a13850f0aa58c3541c9a1e19c1
345e1041a204ca9d40a25b71ae239ce3275f0755

**Comment by paramat** [comment]
Is there a corresponding issue to close?

**Comment by HybridDog** [comment]
AfaIk, there isn't.
